agent/eventhandler/task_handler.go

```diff
@@ -157,7 +157,8 @@ func (handler *TaskHandler) AddStateChangeEvent(change statechange.Event, client
 // startDrainEventsTicker starts a ticker that periodically drains the events queue
 // by submitting state change events to the ECS backend
 func (handler *TaskHandler) startDrainEventsTicker() {
-	derivedCtx, _ := context.WithCancel(handler.ctx)
```

Comment:
`agent/eventhandler/task_handler.go:159: the cancel function returned by context.WithCancel should be called, not discarded, to avoid a context leak`